server/build/webpack.js

```diff
@@ -32,7 +34,7 @@ export default async function createCompiler (dir, { dev = false, quiet = false
   const defaultEntries = dev
     ? [join(__dirname, '..', '..', 'client/webpack-hot-middleware-client')] : []
   for (const p of pages) {
-    entry[join('bundles', p)] = defaultEntries.concat([`./${p}?entry`])
+    entry[pagePathToJson(join('bundles', p))] = defaultEntries.concat([`./${p}?entry`])
```

Comment:
I think, ideally, extensions should be changed on `JsonPagesPlugin`. Is it difficult ?